### PR TITLE
Update RootTopic to support organization id for OtA updates

### PR DESCRIPTION
**Pull request by stevenkuhn**
This adds an additional MQTT topic that a device subscribes to that includes the organization id. That organization id is retrieved from the device's JWT.

### CODE REVIEW

File: source/Meadow.Core/Update/UpdateService.cs [hunk 1]
```diff
@@ -572,6 +580,49 @@ private void DisplayTree(DirectoryInfo di)
         }
     }
 
+    // In order to not require the use of a library, this was taken from
+    // RFC7515, Appendix C: https://datatracker.ietf.org/doc/html/rfc7515#appendix-C
+    private string base64UrlDecode(string arg)
```
Comment (stevenkuhn) [inline source/Meadow.Core/Update/UpdateService.cs:585]: There are libraries that support parsing and validating JWTs like [Microsoft.IdentityModel.JsonWebTokens](https://learn.microsoft.com/en-us/dotnet/api/microsoft.identitymodel.jsonwebtokens?view=msal-web-dotnet-latest), but this seemed like a simpler solution that didn't require additional libraries.

File: source/Meadow.Core/Update/UpdateService.cs [hunk 2]
```diff
@@ -572,6 +580,49 @@ private void DisplayTree(DirectoryInfo di)
...
+        string s = arg;
+        s = s.Replace('-', '+'); // 62nd char of encoding
+        s = s.Replace('_', '/'); // 63rd char of encoding
+        switch (s.Length % 4) // Pad with trailing '='s
+        {
+            case 0: break; // No pad chars in this case
+            case 2: s += "=="; break; // Two pad chars
+            case 3: s += "="; break; // One pad char
+            default:
+                throw new Exception("Illegal base64url string!");
```
Comment (stevenkuhn) [inline source/Meadow.Core/Update/UpdateService.cs:596]: What's the best thing to do with exceptions like this and the exceptions that are thrown below? Should I have better error messages? Should it return null instead and handle null checking on the calling side?

File: source/Meadow.Core/Update/UpdateService.cs [hunk 3]
```diff
@@ -572,6 +580,49 @@ private void DisplayTree(DirectoryInfo di)
...
+        }
+
+        var bytes = Convert.FromBase64String(s);
+        return Encoding.UTF8.GetString(bytes);
+    }
+
+    private JsonWebTokenPayload? GetJsonWebTokenPayload(string? jwt)
+    {
+        if (string.IsNullOrWhiteSpace(jwt))
```
Comment (ctacke) [inline source/Meadow.Core/Update/UpdateService.cs:605]: totally minor, but I think perf is better if you use `ArgumentNullException.ThrowIfNullOrEmpty()`

**Review by ctacke — APPROVED** [review]

File: source/Meadow.Core/Configuration/MeadowUpdateSettings.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ internal class MeadowUpdateSettings : IUpdateSettings
     public string AuthServer { get; set; } = "https://www.meadowcloud.co";
     public int AuthPort { get; set; } = 443;
     public string Organization { get; set; } = "Default organization";
-    public string RootTopic { get; set; } = "ota;ota/{ID}/updates";
+    public string RootTopic { get; set; } = "ota;ota/{ID}/updates;{OID}/ota/{ID}";
```
Comment (lamebrain) [inline source/Meadow.Core/Configuration/MeadowUpdateSettings.cs:13]: Let's remove `ota;ota/{ID}/updates;`. When we move over to EMQX the device won't be able to connect to these anyway.

Comment (stevenkuhn) [reply source/Meadow.Core/Configuration/MeadowUpdateSettings.cs:13]: I can do that. My only concern is that once this is merged in, then we are potentially committed to rolling out Meadow.Cloud on the next release of Meadow.Core. If there's a hotfix that's required (and that hotfix is based on `develop`), then OtA breaks until the changes to Meadow.Cloud are in production.

File: source/Meadow.Core/Update/UpdateService.cs (hunk 3 above)
Comment (stevenkuhn) [reply source/Meadow.Core/Update/UpdateService.cs:605]: From what I can tell `netstandard2.1` does not have `ArgumentNullException.ThrowIfNull()` nor `ArgumentException.ThrowIfNullOrEmpty()`.

File: source/Meadow.Core/Configuration/MeadowUpdateSettings.cs (hunk 4 above)
Comment (stevenkuhn) [reply source/Meadow.Core/Configuration/MeadowUpdateSettings.cs:13]: I'm going to leave the existing RootTopics in until we have this tested further.

**Pull request merged**